src/DependencyInjection/FOSHttpCacheExtension.php

```diff
@@ -430,6 +430,12 @@ private function loadVarnish(ContainerBuilder $container, XmlFileLoader $loader,
         $container->setParameter('fos_http_cache.proxy_client.varnish.options', $options);
 
         $loader->load('varnish.xml');
+
+        $requestFactory = isset($config['http']['request_factory'])
+            ? new Reference($config['http']['request_factory'])
+            : null;
+        $container->getDefinition('fos_http_cache.proxy_client.varnish')
+            ->replaceArgument(2, $requestFactory);
```

Comment:
i would inline this to make things more compact:

```suggestion
            ->replaceArgument(2, $config['http']['request_factory'] ?? null);
```

(also in the other places below)

Reply:
I'd love to do this, but it has to be wrapped in `new Reference(...)` when it's set, and I'm afraid that won't work with the null coalescing operator.

Reply:
you are correct, sorry.